business/MetadataDatabase/Services/AlgoService.cs

```diff
@@ -45,29 +52,45 @@ public string Execute(AlgoExeInfoDto algoExeInfo)
             {
                 dir.Delete(true);
             }
-
+            // Clean workspace algos
+            di = new DirectoryInfo(workspaceAlgosName);
+            foreach (FileInfo file in di.GetFiles())
+            {
+                file.Delete();
+            }
+            foreach (DirectoryInfo dir in di.GetDirectories())
+            {
+                dir.Delete(true);
+            }
+            // Get the specified algo to execute
+            var algoToExecute = this.Get(algoExeInfo.AlgoId);
+            // Create the python algo file to execute
+            File.WriteAllText($"{workspaceAlgosName}{algoToExecute.MainFile}", algoToExecute.ContentFile);
+            this.logger.LogInformation($"Main file {algoToExecute.MainFile}");
+            // Download the series zip file on which the algo will be executed
+            var zipFileName = this.pacsService.DownloadSeries(algoExeInfo.SeriesInstanceUID);
             // Extract zip file in workspace
-            ZipFile.ExtractToDirectory(zipFileName, workspaceName);
+            ZipFile.ExtractToDirectory(zipFileName, workspaceDataName);
             // Get path to dicom files
             ZipArchive archive = ZipFile.OpenRead(zipFileName);
-            var fullworkspacePath = Path.GetFullPath(workspaceName);
-            Console.WriteLine(fullworkspacePath);
-            var relativeFilePath = Path.Join(workspaceName + archive.Entries[0].FullName);
-            Console.WriteLine(relativeFilePath);
+            var fullworkspacePath = Path.GetFullPath(workspaceDataName);
+            this.logger.LogDebug(fullworkspacePath);
```

Comment:
Will these log be usable or should we add some context?

Reply:
No deleted ;-)